jobs/integration/conftest.py

```diff
@@ -301,53 +301,42 @@ async def model(request, tools):
 
 
 @pytest.fixture(scope="module")
-async def k8s_cloud(model, tools):
-    kcp_app = model.applications["kubernetes-control-plane"]
-    kcp_unit = kcp_app.units[0]
-    created_k8s_cloud = False
-
-    with NamedTemporaryFile(dir=Path.home() / ".local" / "share" / "juju") as f:
-        await scp_from(
-            kcp_unit,
-            "config",
-            f.name,
+async def k8s_cloud(kubeconfig, tools):
+    clouds = await tools.run(
+        "juju", "clouds", "--format", "yaml", "-c", tools.controller_name
+    )
+    if tools.k8s_cloud in yaml.safe_load(clouds[0]):
+        yield tools.k8s_cloud
+        return
```

Comment:
first -- let's see if the cloud we want already exists -- to prevent a failure when adding it a second time